### PR TITLE
Dev 114 bmrs eic dukes

### PR DESCRIPTION
**Pull request by msff2**
@mdhillman John and I addressed your following comments.

a) Update the README file (TheWorldAvatar\UK_Power_Generation_Units_BMRS\Python\Code\ukpowergeneration\README.txt) of the UKPowerGeneration project with a generic description.

b) Describe the data source used in the project in the same README file.

c) Provide a screenshot of the visualisation interface in the README file to help future users understand if their generated output looks alike.

### CODE REVIEW

**Comment by mdhillman** [comment]
I doubt I will have to look at this before I leave CMCL, as such I'm adding @gpeb2 and @Ushcode (at least one of them should take a look).

I can already say though that I think a better file location for the project can be found, I'm not a fan of adding new top-level directories to the repository. Perhaps the `/web` directory is a better home?

**Comment by sm453** [comment]
How does this PR relate to dev-migrate-johns-projects-to-stack? Is that going to be discarded? Or will there be a separate PR for that?

**Comment by msff2** [comment]
> How does this PR relate to dev-migrate-johns-projects-to-stack? Is that going to be discarded? Or will there be a separate PR for that?

That is going to be discarded.

**Comment by msff2** [comment]
> I doubt I will have to look at this before I leave CMCL, as such I'm adding @gpeb2 and @Ushcode (at least one of them should take a look).
> 
> I can already say though that I think a better file location for the project can be found, I'm not a fan of adding new top-level directories to the repository. Perhaps the `/web` directory is a better home?

Good point. We will address this.

**Comment by sm453** [comment]
> > How does this PR relate to dev-migrate-johns-projects-to-stack? Is that going to be discarded? Or will there be a separate PR for that?
> 
> That is going to be discarded.

Just for the record: Having discussed it, a preferable action will be to merge dev-migrate-johns-projects-to-stack as part of a separate PR, once this PR is merged.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukexportcurtailmentimporter/resources/ukexportcurtailmentimporter.properties
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukexportcurtailmentimporter/resources/ukexportcurtailmentimporter.properties:10]: Please remove the key value.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-BMRS-API-Input-JA-#-ReadAll.md
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-BMRS-API-Input-JA-#-ReadAll.md:11]: Please update to:
`dropbox\CoMo shared\ja685\Modelling-Resources\BMRS\Script-BMRS-API`

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-Mapping-Match-JA-#.md
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-Mapping-Match-JA-#.md:11]: Please update to:
`dropbox\CoMo shared\ja685\Modelling-Resources\BMRS\Script-BMRS-API`

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/mapping/README-Mapping-Match-JA-#.md
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/mapping/README-Mapping-Match-JA-#.md:12]: Please update to:
`dropbox\CoMo shared\ja685\Modelling-Resources\BMRS\Script-BMRS-API`

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/BMRS_API_Input_JA_7.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/BMRS_API_Input_JA_7.py:1]: Please remove this line.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/BMRS_API_Input_JA_7.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/BMRS_API_Input_JA_7.py:297]: Please remove the API key value from lines like this (there are a few in this file). I would not remove the lines altogether - they might still be useful.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/resources/ukpowergeneration.properties
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/resources/ukpowergeneration.properties:10]: Please remove the key value (not the whole line).

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/visualisation/index.html
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/visualisation/index.html:82]: Please remove the key value (not the whole line).

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Marginal-Seller/BMRS-API-JA-MarginalSeller.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Marginal-Seller/BMRS-API-JA-MarginalSeller.py:1]: Please remove this line.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Marginal-Seller/BMRS-API-JA-MarginalSeller.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Marginal-Seller/BMRS-API-JA-MarginalSeller.py:57]: Please remove the API key value from lines like this (there are a few in this file). Don't remove the lines as a whole.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py:1]: Please remove this line.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py:60]: Please remove the API key value from lines like this. Don't remove the lines as a whole.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py:1]: Please remove this line.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py:56]: Please remove the API key value from lines like this. Don't remove the lines as a whole.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py:1]: Please remove this line.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py
Comment (sm453) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py:59]: Please remove the API key from lines like this (there are a few in this file). Don't remove the lines as a whole.

**Review by sm453 — CHANGES_REQUESTED** [review]
I agree with @mdhillman that this should not be at root level. I'd prefer it in `Agents/`, though (even if it isn't one...).

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/visualisation/index.html
Comment (Ushcode) [inline UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/visualisation/index.html:43]:
```suggestion
	        <script src="./twa-vf/twa-vf.min.js" charset="utf-8"></script>

```

remove this and replace with twavf image

**Comment by Ushcode** [comment]
image


viz looks like this

**Comment by Ushcode** [comment]
It would be better to spin this up as part of a stack and use the latest `twa-vf` rather than copying in the `DTVF` files to a different location. They shouldn't be duplicated and are now two years old

**Comment by Ushcode** [comment]
I've moved the `UK_Power_Generation_Units_BMRS` directory into `/Agents`

**Comment by Ushcode** [comment]
I don't have access to `Dropbox (Cambridge CARES)`. May need to commit to the repo to test this

**Comment by sm453** [comment]
> I don't have access to `Dropbox (Cambridge CARES)`. May need to commit to the repo to test this

Just for the record, please _never ever_ commit data to a source code repository. (Ok, exceptions exist, but I don't believe this is one.)

**Comment by Ushcode** [comment]
> > I don't have access to `Dropbox (Cambridge CARES)`. May need to commit to the repo to test this
> 
> Just for the record, please _never ever_ commit data to a source code repository. (Ok, exceptions exist, but I don't believe this is one.)

'Input-Template-Auto.csv' and 'LocalOnlyBMRSKey.txt' do not sound like data to me. It does not make sense to me to merge code to the public TWA repository that depends on files in a private dropbox folder

File: UK_Power_Generation_Units_BMRS/Python/Code/ukexportcurtailmentimporter/README-ExcelToTimeSeries-TimeStampFunctions.txt
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukexportcurtailmentimporter/README-ExcelToTimeSeries-TimeStampFunctions.txt]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-Mapping-Match-JA-#.md
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/ScriptMapQuery/README-Mapping-Match-JA-#.md]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/mapping/README-Mapping-Match-JA-#.md
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowergeneration/mapping/README-Mapping-Match-JA-#.md]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Curtailment/BMRS-API-JA-PerGenerationUnitCurtailment-Bulk-2.py]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Per-Wind-Unit-Export/BMRS-API-JA-PerGenerationUnit-Bulk-2.py]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py]: Done.

File: UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py
Comment (msff2) [reply UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Price-And-Aggregate-By-Type/BMRS-API-PRICE-TYPE-3-FullTime.py]: Done.

**Review by sm453 — APPROVED** [review]

**Pull request merged**
